lib/editor/components/pattern/CalculateDefaultTimesForm.js

```diff
@@ -95,6 +95,9 @@ export default class CalculateDefaultTimesForm extends Component<Props, State> {
     this.setState({speed: +evt.target.value})
 
   render () {
+    const { activePattern } = this.props
+    const activePatternHasLocations = activePattern.patternLocations.length > 0 || activePattern.patternLocationGroups.length > 0
```

Comment:
PatternStopsPanel.js already has a method `patternHasStops`. I would extract this into a utils file and then use that (negated) here!

Reply:
We discussed this partly off of github, but `patternHasStops` isn't quite right for this purpose, and `patternHaltIsStop` can check individual stops/locations/location groups, but not full patterns. I went ahead and extracted `activePatternHasLocations` into a util!